src/metabase/util/i18n/plural.clj

```diff
@@ -0,0 +1,113 @@
+(ns metabase.util.i18n.plural
+  "Resources for parsing the Plural-Forms header from a translation file and determining which of multiple
+  pluralities to use for a translated string."
+  (:require [clojure.core.memoize :as memoize]
+            [instaparse.core :as insta]))
+
+(def ^:private plural-form-parser
+  "This is a parser for the C-like syntax used to express pluralization rules in the Plural-Forms header in
```

Comment:
This was a lot easier to understand with the example you put in the PR description - maybe include that here?